### PR TITLE
Implementing global secrets

### PR DESCRIPTION
**Pull request by donny-dont**
Ready to go!

@tboerger would really like a review since you did the team secrets and this is an extension of that.

@bradrydzewski is the new endpoint ok?

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -246,6 +248,22 @@ func DeleteTeamSecret(c context.Context, s *model.TeamSecret) error {
...
+func GetGlobalSecretList(c context.Context) ([]*model.TeamSecret, error) {
+	return GetTeamSecretList(c, globalTeamName)
+}
+
+func GetGlobalSecret(c context.Context, name string) (*model.TeamSecret, error) {
+	return GetTeamSecret(c, globalTeamName, name)
+}
+
+func SetGlobalSecret(c context.Context, s *model.TeamSecret) error {
+	return SetTeamSecret(c, s)
```
Comment (bradrydzewski) [inline store/store.go]: is the team secret key set to `__global__` when this function is invoked? or does it need to be set here?

Comment (donny-dont) [inline store/store.go]: Probably needs to be set here. Updating.

File: drone/global.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package main
+
+import "github.com/codegangsta/cli"
+
+var globalCmd = cli.Command{
+	Name:  "org",
```
Comment (tboerger) [inline drone/global.go]: global

File: drone/global.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+	Usage: "manage organizations",
```
Comment (tboerger) [inline drone/global.go]: Manage org?

File: drone/global_secret_list.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	"log"
+
+	"github.com/codegangsta/cli"
+)
+
+var globalSecretListCmd = cli.Command{
+	Name:  "ls",
+	Usage: "list all secrets",
+	Action: func(c *cli.Context) {
+		if err := orgSecretList(c); err != nil {
```
Comment (tboerger) [inline drone/global_secret_list.go]: globalSecretList

File: drone/global_secret_rm.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	"log"
+
+	"github.com/codegangsta/cli"
+)
+
+var globalSecretRemoveCmd = cli.Command{
+	Name:  "rm",
+	Usage: "remove a secret",
+	Action: func(c *cli.Context) {
+		if err := orgSecretRemove(c); err != nil {
```
Comment (tboerger) [inline drone/global_secret_rm.go]: globalSecretRemove

File: router/router.go [hunk 6]
```diff
@@ -74,6 +74,15 @@ func Load(middleware ...gin.HandlerFunc) http.Handler {
 		}
 	}
 
+	global := e.Group("/api/global")
+	{
+		global.Use(session.MustTeamAdmin())
```
Comment (tboerger) [inline router/router.go]: MustAdmin?

File: drone/secret.go [hunk 7]
```diff
@@ -11,3 +20,111 @@ var secretCmd = cli.Command{
...
+	}
+
+	return nil
+
+}
+
+// template for secret list items
+var tmplSecretList = "\x1b[33m{{ .Name }} \x1b[0m" + `
+Images: {{ list .Images }}
+Events: {{ list .Events }}
```
Comment (tboerger) [inline drone/secret.go]: Maybe you also want to add the skip verify flag?

File: client/client_impl.go [hunk 8]
```diff
@@ -28,28 +28,30 @@ const (
...
+	pathJob           = "%s/api/repos/%s/%s/builds/%d/%d"
+	pathLog           = "%s/api/repos/%s/%s/logs/%d/%d"
+	pathKey           = "%s/api/repos/%s/%s/key"
+	pathSign          = "%s/api/repos/%s/%s/sign"
+	pathRepoSecrets   = "%s/api/repos/%s/%s/secrets"
+	pathRepoSecret    = "%s/api/repos/%s/%s/secrets/%s"
+	pathTeamSecrets   = "%s/api/teams/%s/secrets"
+	pathTeamSecret    = "%s/api/teams/%s/secrets/%s"
+	pathGlobalSecrets = "%s/api/global/secrets"
+	pathGlobalSecret  = "%s/api/global/secret/%s"
```
Comment (vaijab) [inline client/client_impl.go]: Isn't that supposed to be `"%s/api/global/secrets/%s"` instead of `"%s/api/global/secret/%s"` ? just to keep it consistent.

**Comment by vaijab** [comment]
@donny-dont is there an issue for this feature? I am more interested in the use case for this, etc. Thanks.

**Comment by tboerger** [comment]
These secrets are getting injected into all builds, while org secrets are limited to organization builds and repo secrets are only specific for a single repository.

**Comment by donny-dont** [comment]
@vaijab its basically a replacement for the `PLUGIN_PARAMS` capability in [Drone 0.4](http://readme.drone.io/setup/settings/#plugin-params:fa1a9aa7bf3e774ccda21a0a7b1222dd). For on prem its really helpful.

**Comment by vaijab** [comment]
@donny-dont absolutely. It's very useful when drone is used with on prem gitlab.

**Comment by donny-dont** [comment]
@tboerger @bradrydzewski tested locally. Think its good to go now.

File: drone/secret.go (hunk 7 above)
Comment (donny-dont) [inline drone/secret.go]: Added

**Comment by tboerger** [comment]
LGTM

**Pull request merged**